Lombiq.Tests.UI/Extensions/VisualVerificationUITestContextExtensions.cs

```diff
@@ -0,0 +1,509 @@
+    /// The reference image is automatically loaded from assembly resource, if it's not exists then from the project
+    /// path based on <see cref="VisualMatchApprovedConfiguration"/> - it can be configured over
+    /// <paramref name="configurator"/> -, if the reference image doesn't exist, a new one will be created based on the
+    /// element's screenshot, and an <see cref="VisualVerificationReferenceImageCreatedException"/> will be thrown. The
+    /// reference image path is generated from the method name - annotated with
+    /// <see cref="VisualVerificationAttribute"/> - and the source file name and path, where the method is.
+    /// </summary>
+    /// <param name="context">The <see cref="UITestContext"/> in which the extension is executed on.</param>
+    /// <param name="element">Target element.</param>
+    /// <param name="comparator">To validate the comparison result.</param>
+    /// <param name="roi">Region of interest. Can be  null.</param>
+    /// <param name="configurator">Action callback to configure the behavior. Can be null.</param>
+    /// <exception cref="VisualVerificationReferenceImageCreatedException">
+    /// If no reference image found under project path.
+    /// </exception>
+    /// <exception cref="VisualVerificationAttributeNotFoundException">
+    /// If no method found annotated with <see cref="VisualVerificationAttribute"/> in stacktrace.
+    /// </exception>
+    private static void VisualAssertApproved(
+        this UITestContext context,
+        IWebElement element,
+        Action<VisualMatchApprovedContext, ICompareResult> comparator,
+        Rectangle? roi = null,
+        Action<VisualMatchApprovedConfiguration> configurator = null)
+    {
+        var configuration = new VisualMatchApprovedConfiguration();
+        configurator?.Invoke(configuration);
+
+        var stackTrace = new EnhancedStackTrace(new StackTrace(fNeedFileInfo: true));
+        var testFrame = stackTrace
+            .FirstOrDefault(frame =>
+                frame.MethodInfo.MethodBase.CustomAttributes
+                    .Any(attribute => attribute.AttributeType == typeof(VisualVerificationAttribute)));
+
+        if (testFrame == null)
+        {
+            throw new VisualVerificationAttributeNotFoundException();
+        }
+
+        var approvedContext = new VisualMatchApprovedContext
+        {
+            ModuleName = testFrame.MethodInfo.DeclaringType.Name,
+            MethodName = testFrame.MethodInfo.Name,
+        };
+
+        approvedContext.ReferenceFileName = configuration.ReferenceFileNameFormatter(
+            approvedContext.ModuleName,
+            approvedContext.MethodName);
+
+        // Try loading reference image from embedded resources first.
+        approvedContext.ReferenceResourceName = $"{testFrame.MethodInfo.DeclaringType.Namespace}.{approvedContext.ReferenceFileName}.bmp";
+        var referenceImage = testFrame.MethodInfo.DeclaringType.Assembly
```

Comment:
This isn't disposed.

Reply:
Yes, It will be disposed at the end of the test with `GenericFailureDumpItem<TContent>`

Reply:
I see this being disposed on line 233, that I missed, so it is actually properly disposed.